reference/src/main/scala/akka/stream/alpakka/reference/settings.scala

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright (C) 2016-2018 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package akka.stream.alpakka.reference
+
+// rename Java imports if the name clashes with the Scala name
+import java.time.{Duration => JavaDuration}
+import java.util.function.Predicate
+
+import scala.concurrent.duration.Duration
+import scala.concurrent.duration._
+
+/**
+ * Settings class constructor is private and not exposed as API.
+ * Adding or removing arguments to methods with default values is not binary
+ * compatible. However, since the constructor is private, it will be possible
+ * to add or remove attributes without introducing binary incompatibilities.
+ */
+final class SourceSettings private (
+    val clientId: Option[String] = None,
+    val authentication: Authentication = Authentication.None,
+    val pollInterval: Duration = 5.seconds
+) {
```

Comment:
another question: where would you actually validate these SourceSettings ? What if , for example, clientId is not optional and absolutely required for the Source to work but the user provides SourceSettings.withAuthentication().withPollInterval()  (and accidentally leaving clientId to None) ???

Reply:
Good point. For such mandatory attributes, we should not have default values and take them as parameters in the `apply` and `create` factory methods. Added a commit that illustrates that.